data/dataexchange/src/main/java/tech/pegasys/teku/data/SlashingProtectionImporter.java

```diff
@@ -129,35 +129,56 @@ private SigningHistory signingHistoryConverter(final SigningHistory signingHisto
     }
   }
 
-  public void updateLocalRecords(final Path slashingProtectionPath) {
+  public void updateLocalRecords(
+      final Path slashingProtectionPath, final Consumer<String> statusConsumer) {
     this.slashingProtectionPath = slashingProtectionPath;
-    data.forEach(this::updateLocalRecord);
-    log.display("Updated " + data.size() + " validator slashing protection records");
+    data.forEach(record -> updateLocalRecord(record, statusConsumer));
+    statusConsumer.accept("Updated " + data.size() + " validator slashing protection records");
   }
 
-  private void updateLocalRecord(final SigningHistory signingHistory) {
+  public Optional<String> updateSigningRecord(final BLSPublicKey publicKey) {
+    final BLSPubKey key = new BLSPubKey(publicKey);
+    data.stream()
+        .filter(signingHistory -> signingHistory.pubkey.equals(key))
+        .collect(Collectors.toList())
+        .forEach(record -> updateLocalRecord(record, (__) -> {}));
+    if (validatorImportErrors.containsKey(publicKey)) {
+      return Optional.of(validatorImportErrors.get(publicKey));
+    }
+    return Optional.empty();
+  }
+
+  private void updateLocalRecord(
+      final SigningHistory signingHistory, final Consumer<String> statusConsumer) {
     String validatorString = signingHistory.pubkey.toBytes().toUnprefixedHexString().toLowerCase();
+    final String hexValidatorPubkey = signingHistory.pubkey.toHexString();
 
-    log.display("Importing " + validatorString);
+    statusConsumer.accept("Importing " + validatorString);
     Path outputFile = slashingProtectionPath.resolve(validatorString + ".yml");
     Optional<ValidatorSigningRecord> existingRecord = Optional.empty();
     if (outputFile.toFile().exists()) {
       try {
         existingRecord = syncDataAccessor.read(outputFile).map(ValidatorSigningRecord::fromBytes);
       } catch (IOException e) {
-        log.exit(1, "Failed to read existing file: " + outputFile);
+        validatorImportErrors.put(
+            signingHistory.pubkey.asBLSPublicKey(), "unable to load existing record.");
+        statusConsumer.accept("Failed to read existing file: " + outputFile);
+        return;
```

Comment:
I think we're getting a significant change in behaviour here for the import subcommand.  Previously this was calling `log.exit` which would immediately stop importing whereas we'd now report the error and move on to trying to import for the next key.

That's probably ok since trying to import as much as we can probably makes sense.  I think we'll get an awful lot of repeated errors if the genesis validators root doesn't match though which is a bit unfriendly.

Reply:
it's true that this is functionally different, but yes the decision to stop was based on not having a better use case, and now i think we have a better use-case this can keep trying rather than die...

Reply:
and yep its ugly when the validators root doesn't match, I did try that specific case...